data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/WithKeysProcessor.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.mutatestring;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.List;
+
+public abstract class WithKeysProcessor extends AbstractStringProcessor {
+    private final List<String> keys;
+
+    public WithKeysProcessor(final PluginMetrics pluginMetrics, final WithKeysProcessorConfig config) {
+        super(pluginMetrics);
+        this.keys = config.getWithKeys();
+    }
+
+    @Override
+    protected void performStringAction(Event recordEvent) {
+        for(String key : keys) {
+            final Object value = recordEvent.get(key, Object.class);
+            if(value instanceof String) {
```

Comment:
It looks like your method checks for a value of string to skip a value that is an unexpected type. If the key is present with a null value it looks like an unhandled runtime exception will still be thrown by `JacksonEvent`. Not sure if this is intended.

Reply:
After checking this scenario it looks like this in handled by jackson